### PR TITLE
🧑🏻 Contact Me modal improvements

### LINKED ISSUE #389: [Contact Me] Update Mentor side Email Received Notification
#### Description
We'll need to update the function `contact_mentor()` that resides in the file `views/messages.py` so that it
- Supports the new information that is taken from the form
- Sends the correct email in the correct format

You'll also need to go into our SendGrid account and update the existing template / design a new email template that supports the new information.

Finally, make sure that you collaborate with #390 issuer so conform with an agreed JSON file

#### Screenshots
[Refer to the Figma](https://www.figma.com/file/9macEfYzayWu0k6ogNOUJA/Mentee-Mockup?node-id=2286%3A14354)

#### Files
`views/messages.py`

### PR DESCRIPTION
**Pull request by nroy20**
## Description :star2:

Fixes #389 and #390

Updated Contact Me modal to include new fields (added preferred method of contact + interest areas); modified sendMenteeMentorEmail() , messages.py endpoint, and SendGrid template to account for changes.

Changed messages.py, MentorContactModal.js, and api.js

### CODE REVIEW

File: frontend/src/components/MentorContactModal.js [hunk 1]
```diff
@@ -39,43 +49,89 @@ function MentorContactModal({ mentorId, menteeId, mentorName }) {
...
-            width={120}
-            content={"Send Message"}
-            onClick={async () => {
-              const res = await sendMenteeMentorEmail(
-                mentorId,
-                menteeId,
-                message
-              );
-              if (!res) {
-                console.log("Failed to send message");
```
Comment (pguduguntla) [inline frontend/src/components/MentorContactModal.js:71]: do we wanna remove these prints statements?

File: frontend/src/components/MentorContactModal.js [hunk 2]
```diff
@@ -39,43 +49,89 @@ function MentorContactModal({ mentorId, menteeId, mentorName }) {
...
+            />
+            <br />
+            {error && (
+              <>
+                <b style={{ color: "red" }}>
+                  Failed to send message, please try again
+                </b>
+                <br />
+              </>
+            )}
```
Comment (pguduguntla) [inline frontend/src/components/MentorContactModal.js:112]: So, in terms of the error message, I actually added a field inside ModalInput where you can pass in an error = True and errorMessage props. I would  consider looking at how I implemented it in EditProfile or any of the registration components to see how to implement it. I think it'll make it more consistent.

File: frontend/src/components/MentorContactModal.js [hunk 3]
```diff
@@ -39,43 +49,89 @@ function MentorContactModal({ mentorId, menteeId, mentorName }) {
...
+                const res = await sendMenteeMentorEmail(
+                  mentorId,
+                  menteeId,
+                  responseEmail,
+                  interestAreas,
+                  communicationMethod,
+                  message
+                );
+                if (!res) {
+                  console.log("Failed to send message");
```
Comment (pguduguntla) [inline frontend/src/components/MentorContactModal.js:126]: keep print statement?

File: package.json [hunk 4]
```diff
@@ -9,5 +9,6 @@
         "setup:backend": "cd backend && poetry install",
         "setup:frontend": "cd frontend && yarn",
         "setup": "concurrently \"yarn setup:backend\" \"yarn setup:frontend\""
-    } 
-}
\ No newline at end of file
+    },
+    "dependencies": {}
```
Comment (pguduguntla) [inline package.json:13]: Remove this change

**Review by pguduguntla — CHANGES_REQUESTED** [review]
Comments above^. Only thing i have is to pass the error/errorMessage into <ModalInput /> as a prop. You'll have to add a custom onChange function as well.

File: frontend/src/components/MentorContactModal.js [hunk 5]
```diff
@@ -39,43 +49,89 @@ function MentorContactModal({ mentorId, menteeId, mentorName }) {
...
+            >
+              {filteredOptions.map((item) => (
+                <Select.Option key={item} value={item}>
+                  {item}
+                </Select.Option>
+              ))}
+            </Select>
+          </Form.Item>
+          <Form.Item
+            label="radio"
```
Comment (zfaizal2) [inline frontend/src/components/MentorContactModal.js:75]: adjust label to reflect the name

File: frontend/src/components/MentorContactModal.js [hunk 6]
```diff
@@ -39,43 +49,89 @@ function MentorContactModal({ mentorId, menteeId, mentorName }) {
...
-              if (!res) {
-                console.log("Failed to send message");
-                setError(true);
-              } else {
-                closeModal();
-              }
-            }}
-          />
-        </div>
+        <h1>Reach Out to {mentorName}</h1>
```
Comment (zfaizal2) [inline frontend/src/components/MentorContactModal.js:52]: modify CSS for modal

image

**Review by zfaizal2 — APPROVED** [review]
simple styling changes, but works fine!

**Review by Lam7150 — APPROVED** [review]

**Pull request merged**